src/test/java/ToDoTest.java

```diff
@@ -0,0 +1,26 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ToDoTest {
+    @Test
+    void dummyTest(){
```

Comment:
Nice work on the comprehensive testing, but perhaps you would like to consider separating the tests into different test methods so that you can identify why the test failed/regressions more easily.